src/test/java/plugins/JobDslPluginTest.java

```diff
@@ -817,6 +820,365 @@ public void should_alert_if_generated_job_changed() {
+        assertThat(firstJob, pageObjectExists());
+        assertThat(secondJob, pageObjectExists());
+    }
+
+    /**
+     * Tests whether a new job with template created by JobDsl exists and includes
+     * same description like the template job when the seed job is build.
+     */
+    @Test
+    public void should_create_new_job_by_using_template() {
+
+        // Arrange
+        String testedJobName = "MyJob";
+        String jobDescription = "My sample despription";
+        Job templateJob = jenkins.jobs.create(FreeStyleJob.class, "Template");
+        templateJob.configure();
+        Control descriptionControl = templateJob.control(by.name("description"));
+        descriptionControl.set(jobDescription);
+        templateJob.save();
+        String jobDslScript = String.format("job('%s') { using('%s') }", testedJobName, templateJob.name);
+        Job seed = createSeedJobWithJobDsl(jobDslScript);
+
+        // Act
+        seed.startBuild().waitUntilFinished();
+
+        // Assert
+        Job job = jenkins.jobs.get(Job.class, testedJobName);
+        job.open();
+        WebElement descriptionElement = job.find(by.xpath("//*[@id=\"description\"]/div[1]"));
+        assertEquals(jobDescription, descriptionElement.getText());
+    }
+
+    /**
+     * Tests whether an existing job is renamed by a new job with previous names attribute regex
+     * created by JobDsl when the seed job is build.
+     */
+    @Test
+    public void should_replace_old_job_with_regex() {
+
+        // Arrange
+        String oldJobName = "MyJob01";
+        String newJobName = "MyJob02";
+        String renameRegex = "/MyJob\\d+/";
+        jenkins.jobs.create(FreeStyleJob.class, oldJobName);
+        String jobDslScript = String.format("job('%s') { previousNames( %s ) }", newJobName, renameRegex);
+        Job seed = createSeedJobWithJobDsl(jobDslScript);
+
+        // Act
+        seed.startBuild().waitUntilFinished();
+
+        // Assert
+        Job job = jenkins.jobs.get(Job.class, newJobName);
+        assertThat(job, pageObjectExists());
+        assertEquals(newJobName, job.name);
+    }
+
+    /**
+     * Tests whether two builds are in progress at the same time of a new job with
+     * concurrent build attribute created by JobDsl when the seed job is build and two
+     * builds are triggered.
+     */
+    @Test
+    public void should_run_two_concurrent_builds() {
+
+        // Arrange
+        String jobName = "MyJob";
+        String jobDslScript = String.format("job('%s') { concurrentBuild(); steps { shell('sleep 10') } }", jobName);
+        Job seed = createSeedJobWithJobDsl(jobDslScript);
+
+        // Act
+        seed.startBuild().waitUntilFinished();
+        Job job = jenkins.jobs.get(Job.class, jobName);
+        Build firstBuild = job.scheduleBuild().waitUntilStarted();
+        Build secondBuild = job.scheduleBuild().waitUntilStarted();
+
+        // Assert
+        assertTrue(firstBuild.isInProgress());
```

Comment:
This test implicitly relies on timing, and hence it may be flaky. OTOH 10-second timeout should be enough